previous-version/previous_version.sh

```diff
@@ -0,0 +1,189 @@
+#!/bin/bash
+#
+# Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+# one or more contributor license agreements. See the NOTICE file distributed
+# with this work for additional information regarding copyright ownership.
+# Licensed under the Camunda License 1.0. You may not use this file
+# except in compliance with the Camunda License 1.0.
+#
+
+# Given a current version identifies a previous version based on existing git tags
+# Params:
+#   1 - Current version. E.g. 8.8.0
+#   2 - Verbose mode (true|false)
+# Output:
+#   "$TYPE $PREV_TAG"
+#   E.g. "NORMAL 8.7.0"
+
+PREV_TAG="not found"
+VERSION=$1
+VERBOSE=$2
+
+if [ "$VERBOSE" = "true" ]; then
+  set -x
+fi
+
+  # Function to determine the previous tag using 'NORMAL' logic
+  determine_normal_prev_tag() {
+      local candidate=""
+      local candidateWithPatch0=""
+      # Loop through all previous tags to find the appropriate previous tag
+      for tag in $prev_tags; do
+          if [[ $tag != "$current_tag" ]] && [[ $tag =~ $NORMAL_PATTERN ]]; then
+              IFS='.' read -r tag_major tag_minor tag_patch <<< "$tag"
+              IFS='.' read -r curr_major curr_minor curr_patch <<< "$current_tag"
+              curr_patch=${curr_patch%%-*}
+               if { ((tag_major == curr_major)) && ((tag_minor == curr_minor)) && ((tag_patch == curr_patch)); }; then
```

Comment:
I'm confused with this syntax and not sure it would work.
What about:
```suggestion
               if (( tag_major == curr_major && tag_minor == curr_minor && tag_patch == curr_patch )); then
```

Reply:
@maxdanilov this is a good question and hard for me to answer, as I don't have much experience with those approaches. Here is what AI thinks about it (**TLDR it prefers the current version as more reliable and explicit**):
<img width="729" alt="image" src="https://github.com/user-attachments/assets/3b4a5f96-4694-402e-bfce-f552c87b46c0" />
<img width="725" alt="image" src="https://github.com/user-attachments/assets/41ff8398-2aee-465d-8b4d-2032656eb0b8" />

Reply:
I'd say direct chaining is easier to read and less confusing, so let's go for it.